src/pages/ConnectWallet/ConnectWallet.tsx

```diff
@@ -11,22 +15,79 @@ import OrbsStatic from 'assets/orbs-static.png'
 import { FoxIcon } from 'components/Icons/FoxIcon'
 import { Page } from 'components/Layout/Page'
 import { RawText, Text } from 'components/Text'
-import { ActionTypes, WalletActions } from 'context/WalletProvider/WalletProvider'
+import { KeyManager, SUPPORTED_WALLETS } from 'context/WalletProvider/config'
+import { ActionTypes, InitialState, WalletActions } from 'context/WalletProvider/WalletProvider'
 import { useQuery } from 'hooks/useQuery/useQuery'
 import { colors } from 'theme/colors'
 
-type NoWalletProps = {
+type WalletProps = {
+  state: InitialState
   dispatch: Dispatch<ActionTypes>
-  hasWallet: boolean
 }
 
-export const ConnectWallet = ({ dispatch, hasWallet }: NoWalletProps) => {
+async function connectCypressWallet(
+  keyring: Keyring,
+  dispatch: Dispatch<ActionTypes>,
+  walletId: string,
+  walletPassword: string
+) {
+  const adapter = SUPPORTED_WALLETS[KeyManager.Native].adapter.useKeyring(keyring)
+  if (adapter) {
+    const wallet = await adapter.pairDevice(walletId)
+    await wallet.initialize()
+  }
+
+  const vaultIds = await Vault.list()
+  for (let index = 0; index < vaultIds.length; index++) {
+    const deviceId = vaultIds[index]
+    if (deviceId !== walletId) {
+      continue
+    }
+
+    const wallet = keyring.get<NativeHDWallet>(deviceId)
+    const vault = await Vault.open(deviceId, decodeURIComponent(walletPassword))
+    const mnemonic = (await vault.get('#mnemonic')) as native.crypto.Isolation.Core.BIP39.Mnemonic
+    mnemonic.addRevoker?.(() => vault.revoke())
+    await wallet?.loadDevice({
+      mnemonic,
+      deviceId
+    })
+    const { name, icon } = SUPPORTED_WALLETS[KeyManager.Native]
+    dispatch({
+      type: WalletActions.SET_WALLET,
+      payload: {
+        wallet,
+        name,
+        icon,
+        deviceId,
+        meta: { label: vault.meta.get('name') as string }
+      }
+    })
+    dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+  }
+}
+
+export const ConnectWallet = ({ state, dispatch }: WalletProps) => {
+  const isCypressTest =
+    localStorage.hasOwnProperty('walletIdCypress') &&
+    localStorage.hasOwnProperty('walletPasswordCypress')
+  const hasWallet = Boolean(state.walletInfo?.deviceId)
   const history = useHistory()
   const translate = useTranslate()
   const query = useQuery<{ returnUrl: string }>()
   useEffect(() => {
```

Comment:
One other issue now is that this effect now triggers 4 times, notably, some after the Dashboard route is loaded, and hence after the tests are started.

This breaks some tests, as it resets the route whilst they are running. A good way to reproduce is to look at the `nav bar works` test, where it'll force the route back to `dashboard` when it is trying to check the nav bar routes.

Reply:
Some filters are applied to mitigate this problem.